src/borg/archiver.py

```diff
@@ -696,43 +696,78 @@ def do_rename(self, args, repository, manifest, key, cache, archive):
-                              str(cache),
-                              DASHES, logger=logging.getLogger('borg.output.stats'))
+        with Cache(repository, key, manifest, lock_wait=self.lock_wait) as cache:
+            archive = Archive(repository, key, manifest, args.location.archive, cache=cache)
+            stats = Statistics()
+            archive.delete(stats, progress=args.progress, forced=args.forced)
+            manifest.write()
+            repository.commit(save_space=args.save_space)
+            cache.commit()
+            logger.info("Archive deleted.")
+            if args.stats:
+                log_multi(DASHES,
+                          stats.summary.format(label='Deleted data:', stats=stats),
+                          str(cache),
+                          DASHES, logger=logging.getLogger('borg.output.stats'))
+        return self.exit_code
+
+    def _delete_archives(self, args, repository):
+        """Delete multiple archives"""
+        if args.location.archive:
+            logger.error('The options --oldest and --latest have no effect on archive targets.')
+            self.exit_code = EXIT_ERROR
         else:
-            if not args.cache_only:
-                msg = []
-                try:
-                    manifest, key = Manifest.load(repository)
-                except NoManifestError:
-                    msg.append("You requested to completely DELETE the repository *including* all archives it may contain.")
-                    msg.append("This repository seems to have no manifest, so we can't tell anything about its contents.")
-                else:
-                    msg.append("You requested to completely DELETE the repository *including* all archives it contains:")
-                    for archive_info in manifest.list_archive_infos(sort_by='ts'):
-                        msg.append(format_archive(archive_info))
-                msg.append("Type 'YES' if you understand this and want to continue: ")
-                msg = '\n'.join(msg)
-                if not yes(msg, false_msg="Aborting.", truish=('YES', ),
-                           env_var_override='BORG_DELETE_I_KNOW_WHAT_I_AM_DOING'):
-                    self.exit_code = EXIT_ERROR
-                    return self.exit_code
-                repository.destroy()
-                logger.info("Repository deleted.")
-            Cache.destroy(repository)
-            logger.info("Cache deleted.")
+            manifest, key = Manifest.load(repository)
+            n = args.oldest or args.latest
+            archives = manifest.list_archive_infos('ts', reverse=args.latest)[:n]
+            for archive in archives:
+                logger.info('Deleting %s' % archive.name)
+                args.location.archive = archive.name
+                self._delete_archive(args, repository, manifest)
+                if self.exit_code:
+                    break
+            else:
+                logger.error('There are no archives.')
```

Comment:
thanks for reminding me that i need to get rid of that misconception i have about `for ... else`.